### PR TITLE
clear code syntax

### PR DESCRIPTION
**Pull request by wjw1758548031**
- [x] Do only one thing
- [x] Non breaking API changes
- [x] Tested

### What did this pull request do?

clear code syntax

### User Case Description

FirstOrCreate() the syntax becomes more concise

### CODE REVIEW

**Review by demoManito — APPROVED** [review]

File: finisher_api.go [hunk 1]
```diff
@@ -326,45 +326,48 @@ func (db *DB) FirstOrCreate(dest interface{}, conds ...interface{}) (tx *DB) {
...
-				}
-			}
 
-			// initialize with attrs, conds
-			if len(db.Statement.attrs) > 0 {
-				result.assignInterfacesToValue(db.Statement.attrs...)
-			}
+	result := queryTx.Find(dest, conds...)
+	if result.Error != nil {
+		tx.Error = result.Error
```
Comment (a631807682) [inline finisher_api.go:332]: missing return

File: finisher_api.go [hunk 2]
```diff
@@ -326,45 +326,48 @@ func (db *DB) FirstOrCreate(dest interface{}, conds ...interface{}) (tx *DB) {
...
+		exprs := tx.Statement.BuildCondition(db.Statement.assigns[0], db.Statement.assigns[1:]...)
+		assigns := map[string]interface{}{}
+		for _, expr := range exprs {
+			if eq, ok := expr.(clause.Eq); ok {
+				switch column := eq.Column.(type) {
+				case string:
+					assigns[column] = eq.Value
+				case clause.Column:
+					assigns[column.Name] = eq.Value
+				default:
```
Comment (a631807682) [inline finisher_api.go:363]: Is default necessary?

**Review by a631807682 — CHANGES_REQUESTED** [review]

File: finisher_api.go (hunk 2 above)
Comment (wjw1758548031) [reply finisher_api.go:363]: default it's not necessary

File: finisher_api.go (hunk 1 above)
Comment (wjw1758548031) [reply finisher_api.go:332]: I make up

**Pull request closed without merging**